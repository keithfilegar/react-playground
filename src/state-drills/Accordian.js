import React from 'react'

class Accordian extends React.Component {
    // static defaultProps ={
    //     sections: []
    // }
    state ={
        buttonIndex: null
    }

    handleButtonClick = (index) => {
        this.setState({buttonIndex: index})
    }

    renderListItems(section, index, buttonIndex) {
        return (
            <li key={index}>
                <button onClick={() => this.handleButtonClick(index)}>
                    {section.title}
                </button>
                {(buttonIndex === index) && <p>{section.content}</p>}
            </li>
        )
    }

    render() {
        const {buttonIndex} = this.state
        const { sections } = this.props
        return (
            <ul className='Accordian'>
                {sections.map((section, index) => 
                    this.renderListItems(section, index, buttonIndex)
                )}
            </ul>
        )
    }
}

export default Accordian;