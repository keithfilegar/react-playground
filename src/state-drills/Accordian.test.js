import React from 'react'
import ReactDOM from 'react-dom'
import Accordian from './Accordian'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Accordian Component', () => {
    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ]

    it('renders without errors', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty given no props', () => {
        const wrapper = shallow(<Accordian />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('renders list items without displaying text', () => {
        const wrapper = shallow(<Accordian sections={sections} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('opens content when button is clicked', () => {
        const wrapper = shallow(<Accordian sections={sections} />);
        wrapper.find('button').at(1).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('only displays content for the clicked button', () => {
        const wrapper = shallow(<Accordian sections={sections} />);
        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})