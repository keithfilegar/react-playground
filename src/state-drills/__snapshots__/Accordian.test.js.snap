// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component only displays content for the clicked button 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component opens content when button is clicked 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders empty given no props 1`] = `
<ul
  className="Accordian"
/>
`;

exports[`Accordian Component renders list items without displaying text 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
